tests/sentry/integrations/gitlab/tasks/test_update_all_project_webhooks.py

```diff
@@ -0,0 +1,523 @@
+        assert "gitlab_webhook_version" not in org_integration.config
+
+        # Verify SLO halt metric was recorded
+        assert mock_record_event.call_count >= 2
+        assert mock_record_event.call_args_list[0][0][0] == EventLifecycleOutcome.STARTED
+        # Last call should be HALTED with NO_REPOSITORIES reason
+        last_call = mock_record_event.call_args_list[-1]
+        assert last_call[0][0] == EventLifecycleOutcome.HALTED
+        assert last_call[0][1] == GitLabWebhookUpdateHaltReason.NO_REPOSITORIES
+
+    @responses.activate
+    def test_task_handles_org_integration_not_found(self):
+        """Test that the task handles missing org integration"""
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            # Delete org integration
+            self.integration.organizationintegration_set.filter(
+                organization_id=self.organization.id
+            ).delete()
+
+        update_all_project_webhooks(
+            integration_id=self.integration.id,
+            organization_id=self.organization.id,
+        )
+
+        # No API calls should be made
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_task_handles_repositories_missing_webhook_config(self):
+        """Test that the task skips repositories without webhook_id or project_id"""
+        with assume_test_silo_mode(SiloMode.REGION):
+            # Repository missing webhook_id
+            repo_no_webhook = self.create_gitlab_repo(
+                name="repo-no-webhook",
+                external_id=104,
+            )
+            repo_no_webhook.config = {"project_id": "104"}
+            repo_no_webhook.save()
+
+            # Repository missing project_id
+            repo_no_project = self.create_gitlab_repo(
+                name="repo-no-project",
+                external_id=105,
+            )
+            repo_no_project.config = {"webhook_id": "webhook-5"}
+            repo_no_project.save()
+
+        # Mock webhook update for valid repositories
+        responses.add(
+            responses.PUT,
+            "https://example.gitlab.com/api/v4/projects/101/hooks/webhook-1",
+            json={"id": "webhook-1"},
+            status=200,
+        )
+        responses.add(
+            responses.PUT,
+            "https://example.gitlab.com/api/v4/projects/102/hooks/webhook-2",
+            json={"id": "webhook-2"},
+            status=200,
+        )
+        responses.add(
+            responses.PUT,
+            "https://example.gitlab.com/api/v4/projects/103/hooks/webhook-3",
+            json={"id": "webhook-3"},
+            status=200,
+        )
+
+        update_all_project_webhooks(
+            integration_id=self.integration.id,
+            organization_id=self.organization.id,
+        )
+
+        # Only valid repositories should be updated (3 out of 5)
+        assert len(responses.calls) == 3
+
+        # Webhook version should NOT be updated because not all repositories succeeded
```

Comment:
This behavior is interesting. I could see us leaving this as "best effort" vs a guarantee that we updated everything. It's possible that we'll just lack perms to properly update certain configured repositories indefinitely, and this seems like a lot of extra work if users are toggling sync settings.